qdrant-landing/content/documentation/guides/security.md

```diff
@@ -132,6 +130,209 @@ export QDRANT__SERVICE__READ_ONLY_API_KEY=your_secret_read_only_api_key_here
+```json
+{ "Authorization": "Bearer <JWT>" }
+// or 
+{ "Api-Key": "<JWT>" }
+```
+
+#### Generating JSON Web Tokens
+
+Anyone who knows the `api_key` can generate tokens.
+
+- **JWT Header** - Qdrant uses the `HS256` algorithm to decode the tokens.
+
+  ```json
+  {
+    "alg": "HS256",
+    "typ": "JWT"
+  }
+  ```
+
+- **JWT Payload** - You can include any combination of the [claims available](#jwt-payload-claims) in the payload. Keep reading for more info on each one.
+
+  ```json
+  {
+    "exp": 1640995200, // Expiration time
+    "value_exists": ..., // Validate this token by looking for a point with a payload value
+    "access": "r", // Define the access level.
+  }
+  ```
+
+- **JWT Signature** - The signature is generated by encoding the header and payload with the `api_key` using the `HS256` algorithm.
+
+  ```
+  HMACSHA256(
+    base64UrlEncode(header) + "." +
+    base64UrlEncode(payload),
+    api_key
+  )
+  ```
+
+#### JWT Payload Claims
+
+- **`exp`** - The expiration time of the token. This is a Unix timestamp in seconds. The token will be invalid after this time.
+
+  ```json
+  {
+    "exp": 1640995200, // Expiration time
+  }
+  ```
+
+- **`value_exists`** - This is a claim that can be used to validate the token at the cost of an extra lookup.
+  
+  ```json
+  {
+    "value_exists": {
+      "collection": "my_validation_collection",
+      "matches": [
+        { "key": "my_key", "value": "value_that_must_exist" }
+      ],
+    },
+  }
+  ```
+
+  If this claim is present, Qdrant will check if there is a point in the collection with the specified key-values. If it does, the token is valid.
+  
+  This is useful when you need stateful tokens which can be invalidated independently.
+
+- **`access`** - This claim defines the [access level](#table-of-access) of the token. If this claim is present, Qdrant will check if the token has the required access level to perform the operation. If this claim is **not** present, **manage** access is assumed.
+
+  It can provide global access with `r` for read-only, or `m` for manage. For example:
+
+  ```json
+  {
+    "access": "r"
+  }
+  ```
+
+  It can also be specific to one or more collections. The `access` level for each collection is `r` for read-only, or `rw` for read-write, like this:
+
+  ```json
+  {
+    "access": [
+      {
+        "collection": "my_collection",
+        "access": "rw"
+      }
+    ]
+  }
+  ```
+
+  You can also specify which subset of the collection the user is able to access by specifying a `payload` restriction that the points must have.
+
+  ```json
+  {
+    "access": [
+      {
+        "collection": "my_collection",
+        "access": "r",
+        "payload": {
+          "user_id": "user_123456"
+        }
+      }
+    ]
+  }
+  ```
+
+  This `payload` claim will be used to implicitly filter the points in the collection, so that they only see their user, for example.
```

Comment:
Maybe add this explicitly:

    This is equivalent of appending this filter to each request:
    ```json
    {"filter": {"must": [{"key": "user_id", "match": {"value": "user_123456"}}]}}
    ```